### PR TITLE
fix(accessibility): MM-53999 post menu and reminder keyboard support

### PR DESCRIPTION
**Pull request by saturninoabril**
#### Summary
This PR fixes keyboard support via Tab and Arrow Up/Down to access More and Reminder menu of a post.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-53999

#### Screenshots
Screen recording to access More menu (dot menu) via tab and arrow up/down keys.

https://github.com/mattermost/mattermost/assets/5334504/27d9daa2-67b3-4f26-8fc2-a40942100f34

Playwright/Axe test:
Screenshot 2023-08-10 at 12 46 56 PM

#### Release Note
```release-note
* Accessibility tab support to post and reminder menus
```

### CODE REVIEW

File: webapp/channels/src/components/menu/menu_item.tsx [hunk 1]
```diff
@@ -240,7 +240,7 @@ const MenuItemStyled = styled(MuiMenuItem, {
                 '&.Mui-focusVisible .label-elements>:last-child, &.Mui-focusVisible .label-elements>:first-child, &.Mui-focusVisible .label-elements>:only-child': {
                     color: isDestructive && 'var(--button-color)',
                 },
-                '&.Mui-focusVisible .leading-element': {
+                '&.Mui-focusVisible .leading-element, &.Mui-focusVisible .trailing-elements': {
```
Comment (saturninoabril) [inline webapp/channels/src/components/menu/menu_item.tsx:243]: This fixes the styling of delete button on keyboard down (arrow up/down).

From:
Screenshot 2023-08-10 at 12 37 24 PM

To:
Screenshot 2023-08-10 at 12 36 41 PM

**Comment by lindalumitchell** [comment]
@saturninoabril I triggered e2es out of habit, but of course I see you added Playwright coverage. 🎉 So have you already covered e2es for this PR?

**Comment by lindalumitchell** [comment]
Verified on the test server that arrows and Tab work to access a post's More (dot) menu, including the submenu for Reminder. Double-checked the other menu options and also a system message that only has the Delete option. No issues found. 👍

**Comment by saturninoabril** [comment]
Thanks Linda and yes, the Playwright test in this PR covered the accessibility support/change in this PR.

File: webapp/channels/src/components/menu/menu.tsx [hunk 2]
```diff
@@ -176,7 +176,8 @@ export function Menu(props: Props) {
                 aria-label={props.menuButton?.['aria-label'] ?? ''}
                 className={props.menuButton?.class ?? ''}
                 onClick={handleMenuButtonClick}
-                onMouseDown={handleMenuButtonMouseDown}
+                onMouseDown={handleDownEvent}
+                onKeyDown={handleDownEvent}
```
Comment (hmhealey) [inline webapp/channels/src/components/menu/menu.tsx:180]: What are these changes for? It seems like we would want the focus-visible class to be applied when navigating the menu with the keyboard

File: webapp/channels/src/components/dot_menu/post_reminder_submenu.tsx [hunk 3]
```diff
@@ -136,14 +143,18 @@ function PostReminderSubmenu(props: Props) {
                 key={`remind_post_options_${postReminder}`}
                 labels={labels}
                 trailingElements={trailingElements}
-                onClick={() => handlePostReminderMenuClick(postReminder)}
+                onClick={() => handlePostReminder(postReminder)}
+                onKeyDown={(event) => handleKeydown(event, postReminder)}
```
Comment (hmhealey) [inline webapp/channels/src/components/dot_menu/post_reminder_submenu.tsx:147]: Are you sure this is needed? MUI is supposed to fire the `onClick` when an item is selected by pressing space. That's how it works in the channel sidebar at least

File: webapp/channels/src/components/dot_menu/post_reminder_submenu.tsx [hunk 4]
```diff
@@ -136,14 +143,18 @@ function PostReminderSubmenu(props: Props) {
...
             />
         );
     });
 
+    const header = formatMessage({id: 'post_info.post_reminder.sub_menu.header', defaultMessage: 'Set a reminder for:'});
```
Comment (hmhealey) [inline webapp/channels/src/components/dot_menu/post_reminder_submenu.tsx:152]: Minor nit, but could you move the `.replace` call up here? I think it helps to keep the code clear by having those both in the same place

**Comment by hmhealey** [comment]
Regarding this ticket, the tests are good, but I'm actually worried about the fact that we need code changes here to begin with. These new menus had keyboard support in the channel sidebar when they were added, and it seems like that's been broken since then. I don't know if that worked in the dot menu, but I definitely spent a bunch of time testing the channel sidebar menus which now aren't working any more.

**Comment by hmhealey** [comment]
While testing this earlier, menus had proper keyboard support in the latest commit of release-7.11, but not on the latest commit of release-8.0. If I revert the 8.0 cherry-pick of https://github.com/mattermost/mattermost/pull/23749, that seemed to fix it, but reverting it on master didn't seem to fix it unfortunately.

I set up a branch (https://github.com/mattermost/mattermost/compare/hh_revert-menu-a11y) with these new tests and the revert of #23749, but since that doesn't seem to have fixed the keyboard support, they're currently failing

**Comment by saturninoabril** [comment]
Thanks for all the comments and in-depth look at the issues. As discussed offline, I'm fine closing this in favor of your upcoming PR. Please loop me in so I could check Accessibility issues raised by a customer.

**Pull request closed without merging**